controllers/cluster_controller.go

```diff
@@ -243,6 +244,14 @@ func (r *ClusterReconciler) listChildren(ctx context.Context, cluster *clusterv1
 	}
 	controlPlaneMachines, machines := splitMachineList(allMachines)
 
+	secrets := &corev1.SecretList{}
+	secretsListOptions := []client.ListOption{
+		client.InNamespace(cluster.Namespace),
+	}
+	if err := r.Client.List(ctx, secrets, secretsListOptions...); err != nil {
```

Comment:
We can't delete all secrets in the namespace!

Reply:
`reconcileDelete` does a check for OwnerRef before deleting though?

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/cluster_controller.go#L176-L194

Reply:
Oh, right (but it's in `eachFunc`)